modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java

```diff
@@ -318,4 +318,126 @@ public void testDeleteBrandingPreferenceByLocaleQueryParam() throws IOException
                 .assertThat()
                 .statusCode(HttpStatus.SC_NO_CONTENT);
     }
+
+    @Test
+    public void testAddCustomTextPreference() throws IOException, JSONException {
+
+        String body = readResource("add-custom-text.json");
+        Response response = getResponseOfPost(CUSTOM_TEXT_API_BASE_PATH, body);
+
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED)
+                .header(HttpHeaders.LOCATION, notNullValue());
+        String location = response.getHeader(HttpHeaders.LOCATION);
+        assertNotNull(location);
+
+        JSONObject expectedPreference = new JSONObject(new JSONObject(readResource("add-custom-text.json")).
+                get("preference").toString());
+        JSONObject receivedPreference = new JSONObject(new JSONObject(response.asString()).
+                get("preference").toString());
+        Assert.assertTrue(areJSONObjectsEqual(expectedPreference, receivedPreference),
+                "The custom text preference schema of the Response is incorrect");
```

Comment:
```suggestion
                "The custom text preference schema of the Response is incorrect.");
```

Reply:
Fixed with https://github.com/wso2/product-is/pull/16854/commits/f4920b1c7614e08eb8ca0aa4b15907e0c83be572